internal/api/scim/integration_test/users_delete_test.go

```diff
@@ -0,0 +1,91 @@
+//go:build integration
+
+package integration_test
+
+import (
+	"context"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/zitadel/zitadel/internal/integration"
+	"github.com/zitadel/zitadel/internal/integration/scim"
+	"github.com/zitadel/zitadel/pkg/grpc/user/v2"
+	"google.golang.org/grpc/codes"
+	"net/http"
+	"testing"
+)
+
+func TestDeleteUser_errors(t *testing.T) {
+	tests := []struct {
+		name        string
+		ctx         context.Context
+		userID      string
+		errorStatus int
+	}{
+		{
+			name:        "not authenticated",
+			ctx:         context.Background(),
+			errorStatus: http.StatusUnauthorized,
+		},
+		{
+			name:        "no permissions",
+			ctx:         Instance.WithAuthorization(CTX, integration.UserTypeNoPermission),
+			errorStatus: http.StatusNotFound,
+		},
+		{
+			name:        "unknown user id",
+			userID:      "unknown-id",
+			errorStatus: http.StatusNotFound,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			ctx := tt.ctx
+			if ctx == nil {
+				ctx = CTX
+			}
+
+			userID := tt.userID
+			if userID == "" {
```

Comment:
Instead of creating a user, which you can then not remove as the context has no permission, wouldn't it be better to just add a dependency to the test?
Maybe have a look into our user service `internal/api/grpc/user/v2integration_test`

Reply:
This would lead to code duplication (for 2 of the three cases the same user is created). In other endpoints the ratio will be even higher. Usually the only exception is the "not found" case. I'm open to adjusting if you have strong opinions.

Reply:
True, in this instances we use gofakeit to generate the necessary data. Not that I have a particularly strong opinion here, just that creating a user which is never used as the permission check should fail before makes not that much sense to me.

Reply:
The intention was to have a completely valid request, except for the authentication part, to ensure that the request fails solely due to that. I have mixed opinions on whether this approach makes sense. On the one hand, I see the advantage of testing the authentication errors in isolation, as they are not influenced by any other request validation errors. On the other hand, since we are asserting the status codes, I believe it is acceptable to simplify the tests here. I have adjusted the code to use a constant non-existent user ID.